pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -470,6 +470,19 @@ func (ds *Plugin) CreateFederationRelationship(ctx context.Context, fr *datastor
 	})
 }
 
+// DeleteFederationRelationship deletes the federation relationship to the
+// given trust domain. The associated trust bundle is not deleted.
```

Comment:
I didn't include the bundle deletion in the datastore layer, but I am happy to add it if we decide it is a good idea. (See [comment](https://github.com/spiffe/spire/issues/2507#issuecomment-920940233))